debian/nautilus.postinst

```diff
@@ -0,0 +1,34 @@
+#!/bin/sh
+
+XDG_MIMEAPPS_FILE=/etc/xdg/mimeapps.list
+XDG_MIMEAPPS_HEADER="[Default Applications]"
+
+if [ "$1" = configure ]; then
+    if  [ -f ${XDG_MIMEAPPS_FILE} ]; then
+        cp -f ${XDG_MIMEAPPS_FILE} ${XDG_MIMEAPPS_FILE}.dpkg-old
+    fi
+
+    # First non-empty line should be the required header, otherwise the file
+    # will be considered corrupt and will be removed to be filled later.
+    if [ -f ${XDG_MIMEAPPS_FILE}.dpkg-old ]; then
+        sed -e "/^[ \t]*$/d" ${XDG_MIMEAPPS_FILE}.dpkg-old > ${XDG_MIMEAPPS_FILE}.dpkg-new
+        if [ "$(head -n 1 ${XDG_MIMEAPPS_FILE}.dpkg-new)" != "${XDG_MIMEAPPS_HEADER}" ]; then
+            # File found "corrupt", remove it so that we can check later.
+            rm -f ${XDG_MIMEAPPS_FILE}.dpkg-new
```

Comment:
I almost feel like you should bail here since who knows what's in there that you're removing.

Reply:
You will always have the dpkg-old file as a backup in that case, right?

Reply:
...aaand I changed my mind again: I think you're right, better signal the error so that we can catch if something requires attention, instead of silently "fixing things up". Will change it too